os/fs/smartfs/smartfs_utils.c

```diff
@@ -2135,22 +2134,27 @@ ssize_t smartfs_append_data(FAR struct smartfs_mountpt_s *fs, FAR struct smartfs
 		buflen -= readwrite.count;
 		byteswritten += readwrite.count;
 
-		/* Test if we wrote a full sector of data */
+		/* Test if we wrote a full sector of data and more data is pending to be written.
+		 * If more data is not pending, the last modified sector will be synced during close-time
```

Comment:
Small suggestion, I think "If data to be written is not pending" sounds better compared to "If more data is not pending"